imagepipeline/src/main/java/com/facebook/imagepipeline/core/ProducerSequenceFactory.java

```diff
@@ -192,8 +192,12 @@ private static void validateEncodedImageRequest(ImageRequest imageRequest) {
     } else if (UriUtil.isLocalResourceUri(uri)) {
       return getLocalResourceFetchSequence();
     } else {
+      String uriString = uri.toString();
+      if (!TextUtils.isEmpty(uriString)) {
+        uriString.substring(0, Math.min(30, uriString.length));
```

Comment:
Okay, because the error message below was not worded correctly, I misunderstood the code. Given that we want to output the uri scheme, first 30 characters is fine.

Instead of using `isEmpty` and then `Math.min`, it is better to do `if (uri != null && uri.length > 30)`.